### ISSUE TITLE
Extreme slow rewriting of files on NFS drive

### ISSUE THREAD

**Issue by crankedguy** [quodlibet/mutagen#553]
Hello,

I noticed that there are extremely slow speeds when files have to be rewritten. The drive is an NFS mount and a 200MB FLAC needs over 30 seconds to be rewritten. Network  transfer speeds are capped at 30MB/s shown during the operation.
Normal speeds are shown around 125/110 read/write
Any idea why this is the case with Mutagen, I have to rewrite several TB and this cannot be done so.

Thanks

**Comment by lazka** [comment]
Likely because the rewrite is done in blocks: https://github.com/quodlibet/mutagen/blob/399513b167ed00c4b7a9ef98dfe591a276efb701/mutagen/_util.py#L722

**Comment by crankedguy** [comment]
I've seen you just answered this :) Will try it out giving it some more buffer, thanks

**Comment by lazka** [comment]
Otherwise you can use the following from https://mutagen.readthedocs.io/en/latest/user/padding.html

```python
def no_new_padding(info):
    # this will use existing padding but never add new one
    return max(info.padding, 0)

f.save(padding=no_new_padding)
```

This means it will only rewrite the file if absolutely needed. By default it will add more padding the first time to make future changes faster.

**Comment by crankedguy** [comment]
Yes this is the case, most of the files have no cover art at all and if, it has to be updated, also they are not padded well enough by the creator. I will never understand why someone doesn't pad a few bytes nowadays when creating a 2-300MB in the first place. But it is how it is

**Comment by crankedguy** [comment]
I notched this up to 2**21 until now and it brought tremendous speed improvements. Still slow though in terms to normal speed of the shares.

**Comment by phw** [comment]
There is also related #587
